contributors/design-proposals/service-injection-policy.md

```diff
@@ -0,0 +1,444 @@
+# Service Injection Policy
+
+  * [Abstract](#abstract)
+  * [Motivation](#motivation)
+  * [Constraints and Assumptions](#constraints-and-assumptions)
+  * [Use Cases](#use-cases)
+    * [Loose coupling between services and their consumers](#loose-coupling-between-services-and-their-consumers)
+  * [Proposed Changes](#proposed-changes)
+    * [ServiceInjectionPolicy API object](#serviceinjectionpolicy-api-object)
+      * [Validations](#validations)
+    * [AdmissionControl Plug-in: ServiceInjectionPolicy](#admissioncontrol-plug-in-serviceinjectionpolicy)
+      * [Behavior](#behavior)
+      * [Optional Behavior with implicit injections](#optional-behavior-with-implicit-injections)
+  * [Examples](#examples)
+    * [Simple Pod Spec Example](#simple-pod-spec-example)
+    * [Pod Spec with `ConfigMap` Example](#pod-spec-with-`configmap`-example)
+    * [ReplicaSet with Pod Spec Example](#replicaset-with-pod-spec-example)
+
+
+## Abstract
+
+Describes a policy resource that allows loose coupling from services to the pods
+that consume them.
+
+## Motivation
+
+Consuming a service involves more than just connectivity.  In addition to
+coordinates to reach the service, credentials and non-secret configuration
+parameters are typically needed to use the service.  The primitives for this
+already exist, but a gap exists where loose coupling is desired: it should be
+possible to inject pods with the information they need to use a service on a
+service-by-service basis, without the pod authors having to incorporate the
+information into every pod spec where it is needed.
+
+## Constraints and Assumptions
+
+1.  Future work might require new mechanisms to be made to work with controllers
+    such as deployments and replicasets that create pods. Existing controllers
+    that create pods should recreate their pods when a new Service Injection
+    Policy is added that would effect them.
+
+## Use Cases
+
+1.  As a user, I want to be able to describe a way that pods should be injected
+    with the information to consume a particular service. This would allow me
+    to concisely model the information about how the service
+    should be consumed without altering every consuming pod spec.
+
+### Loose coupling between services and their consumers
+
+Using a Service Injection Policy allows pod template authors to not have to explicitly
+set information for every pod. This way authors of pod templates consuming a
+specific service do not need to know all the details about that service.
+
+The information will be injected based off the policy's `Selector`.
```

Comment:
This may be confusing for a newbie since the info isn't influence by the Selector, rather which pods are. Perhaps:
```
The pods chosen for injection will be based off the policy's `Selector`.
```